src/main/java/ch/naviqore/gtfs/schedule/GtfsScheduleParser.java

```diff
@@ -81,13 +78,10 @@ private void parseCalendarDate(CSVRecord record) {
     }
 
     private void parseStop(CSVRecord record) {
-        if (record.isMapped("parent_station")) {
```

Comment:
Why is this check removed?

I would not bypass this and solely rely on the builder internals, which checks for empty parent ids. This is something, which could change in future. If a stop has no parent, I would explicitly call the corresponding method from the builder.

Reply:
This check is now handled through the Utils.getStringFieldOrDefault() method.

Reply:
But still, the method on the builder is called for the case with a parent. If it is not mapped, so the value is null, the builder method for a stop with no parent should be called. So that the logic is not depending on the builder internals, which might change.

Reply:
I really don't see how the builder will handle this case differently in the future and if this would be so, I would handle it inside the parser utils method. Else we'll end up having a method in parseStop that check if(record.isMapped("x") && record.isMapped("y")) { ... } else if ( record.isMapped("x") && ! record.isMapped("y")) {...} ... and so on.

In the new implementation the stop already has three optional values, so you would need to the presence of all resulting in 8 different combinations.

Old Setup in parser:
```
private void parseStop(CSVRecord record) {
    if (record.isMapped("parent_station")) {
        builder.addStop(record.get("stop_id"), record.get("stop_name"), Double.parseDouble(record.get("stop_lat")),
                Double.parseDouble(record.get("stop_lon")), record.get("parent_station").trim());
    } else {
        builder.addStop(record.get("stop_id"), record.get("stop_name"), Double.parseDouble(record.get("stop_lat")),
                Double.parseDouble(record.get("stop_lon")));
    }
}
```

Old Setup in builder:
```
    public GtfsScheduleBuilder addStop(String id, String name, double lat, double lon) {
        addStop(id, name, lat, lon, "");
        return this;
    }

    public GtfsScheduleBuilder addStop(String id, String name, double lat, double lon, String parentStopId) {
        checkNotBuilt();
        if (stops.containsKey(id)) {
            throw new IllegalArgumentException("Stop " + id + " already exists");
        }
        log.debug("Adding stop {}", id);
        Stop stop = new Stop(id, name, new GeoCoordinate(lat, lon));

        // only add stop id if it is not a blank string
        if (!parentStopId.isEmpty()) {
            parents.computeIfAbsent(parentStopId, ignored -> new ArrayList<>()).add(stop);
        }

        stops.put(id, stop);
        return this;
    }
```

Reply:
OK